virl2_client/models/node.py

```diff
@@ -406,15 +410,48 @@ def boot_disk_size(self, value: int) -> None:
 
     @property
     def configuration(self) -> str | None:
-        """Return the initial configuration of this node."""
-        self.lab._sync_topology(exclude_configurations=False)
-        return self._configuration
+        """Return the contents of the main configuration file."""
+        self.lab.sync_topology_if_outdated(exclude_configurations=False)
+        return self._configuration[0].get("content") if self._configuration else None
 
     @configuration.setter
-    def configuration(self, value) -> None:
-        """Set the initial configuration of this node."""
+    def configuration(self, value: str | list | dict) -> None:
+        """
+        Set the content of:
+         - the main configuration file if passed a string,
+         - one configuration file if passed a dictionary in the format of
+        `{"name": "filename.txt", "content": "<file content>"}`,
+         - or multiple configuration files if passed a list of above dictionaries.
+        Can also use "Main" in place of the filename of the main configuration file.
+
+        :param value: The configuration data in one of three formats.
+        """
         self._set_node_property("configuration", value)
-        self._configuration = value
+        self._configuration = (
+            self._configuration if self._configuration is not None else []
+        )
```

Comment:
this can be changed to the following and moved below the `if isinstance(value, str):` block. btw when `self._configuration` can be set to None? for a local node only?
```
if self._configuration is None:
    self._configuration = []
```

Reply:
Oof, brainfart. Thanks.
`Lab.create_node` does:
```python
        for key in ("image_definition", "configuration"):
            if key not in kwargs:
                kwargs[key] = None
```
So when the user does not specify the config when creating the node, and then tries to set config without syncing.

Reply:
```
         if isinstance(value, str):
             if not self._configuration:
 >               self._configuration.append({"name": "Main", "content": value})
 E               AttributeError: 'NoneType' object has no attribute 'append'
 ```
 ...embarrassing. Will have to reopen this.

Reply:
ups, sorry about that ... at least I have seen a proof that config can really be None :D